lib/metadata/event-debouncer.js

```diff
@@ -0,0 +1,114 @@
+"use strict";
+
+var utils = require('../utils');
+
+/**
+ * Debounce protocol events by acting on those events with a sliding delay.
+ * @param {Number} delay
+ * @ignore
+ * @constructor
+ */
+function EventDebouncer(delay) {
+  this._delay = delay;
+  this._queue = null;
+  this._timeout = null;
+}
+
+/**
+ * Adds a new event to the queue and moves the delay.
+ * @param {{ handler: Function, all: boolean|undefined, keyspace: String|undefined,
+ *  cqlObject: String|undefined, callback: Function|undefined }} event
+ */
+EventDebouncer.prototype.eventReceived = function (event) {
+  event.callback = event.callback || utils.noop;
+  this._queue = this._queue || { callbacks: [], keyspaces: {} };
+  //TODO: Warn when size is too large
```

Comment:
In the java-driver there is a differentiation between max pending and max queue size, and the max pending is configurable.  I'm not sure if we need that level of configurability, but I agree that a warning on max queue depth is probably a good idea.